jarviscli/plugins/reminder.py

```diff
@@ -23,11 +23,124 @@
 """
 
+class TagBase:
+    save_key = "reminder_tags"
+    next_id_key = "reminder_tags_next_id"
+
+    def load_tags(self, jarvis):
+        tags = jarvis.get_data(self.save_key)
+        if tags is None:
+            tags = []
+            jarvis.add_data(self.save_key, [])
+
+        return tags
+    
+    def save_tags(self, jarvis, tags):
+        jarvis.update_data(self.save_key, tags)
+    
+    def add_tag(self, jarvis, name):
+        data = self.load_tags(jarvis)
+        new_tag = {
+            "id": self.get_next_id(jarvis),
+            "name": name
+        }
+        data.append(new_tag)
+        self.save_tags(jarvis, data)
+
+    def get_tag_by_id(self, jarvis, id):
+        tags = self.load_tags(jarvis)
+        final_tag = None
+        for tag in tags:
+            if tag["id"] == id:
+                final_tag = tag
+        
+        return final_tag
+
+    def get_next_id(self, jarvis):
+        next_id = jarvis.get_data(self.next_id_key)
+        if next_id is None:
+            next_id = 0
+            jarvis.add_data(self.next_id_key, next_id)
+
+        jarvis.update_data(self.next_id_key, next_id + 1)
+        return next_id
+
+    def clean_up_entry(self, jarvis, entry):
+        id = entry["id"]
+        todos = TodoBase().get_data(jarvis)
+        for todo in todos:
+            new_tags = []
+            for tag_id in todo["tags"]:
+                if tag_id != id:
+                    new_tags.append(tag_id)
+            todo["tags"] = new_tags
+        TodoBase().save_data(jarvis, todos)
+    
+    def format(self, jarvis, tag):
+        return tag["name"]
+    
+    def do_print(self, jarvis):
+        tags = self.load_tags(jarvis)
+        for tag in tags:
+            jarvis.say(self.format(jarvis, tag))
+    
+    def select_one_tag(self, jarvis):
+        data = self.load_tags(jarvis)
+        ask_str = []
+        for entry in data:
+            ask_str.append(self.format(jarvis, entry))
+
+        if len(ask_str) == 0:
+            return None
+
+        title = 'Please choose from tag list (j and k to move up and down, enter to select):'
+        _, index = pick(ask_str, title)
+
+        return data[index]
+
+    def remove(self, jarvis, s):
+        data = self.load_tags(jarvis)
+        if len(data) == 0:
+            jarvis.say("Nothing to remove!")
+            return
+
+        if s.startswith("everything") or s.startswith("all"):
+            for entry in data:
+                self.clean_up_entry(jarvis, entry)
+
+            data = []
+            self.save_data(jarvis, data)
+
+            jarvis.say("ok")
+            return
+
+        # open selection menu
+        ask_str = []
+        for entry in data:
+            ask_str.append(self.format(jarvis, entry))
+
+        title = 'Please choose tag to remove (select with space, j and k to move up and down)'
+        selected = pick(ask_str, title, multi_select=True)
+        print(selected)
```

Comment:
Why this print?

Reply:
forgot to delete some prints that were for testing purposes